java/client/src/main/java/glide/api/models/commands/StreamAddOptions.java

```diff
@@ -0,0 +1,144 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.StreamBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Builder;
+
+/**
+ * Optional arguments to {@link StreamBaseCommands#xadd}
+ *
+ * @see <a href="https://redis.io/commands/xadd/">redis.io</a>
+ */
+@Builder
+public final class StreamAddOptions {
+
+    public static final String NO_MAKE_STREAM_REDIS_API = "NOMKSTREAM";
```

Comment:
Why are these public will the user need to access them?

Reply:
they can if they want, no danger if they do... but for tests.